components/raftstore/src/store/fsm/apply.rs

```diff
@@ -2308,12 +2314,12 @@ where
                 Ok(None) => (),
                 Ok(Some(state)) => {
                     if replace_regions.get(region_id).is_some() {
-                        // This peer must be the first one on local store. So if this peer is created on the other side,
-                        // it means no `RegionLocalState` in kv engine.
+                        // It's marked replaced, then further destroy will skip cleanup, so there
+                        // should be no region local state.
```

Comment:
What if the destroy process happened before updating `pending_create_peers`?

Reply:
It depends. If destroy happen before being [marked as replaced](https://github.com/tikv/tikv/pull/12386/files#diff-ee7d1a7f274d36b83db93153af39a61497eaea746565e713d02b91d68f25a180R2281), then it won't be marked as replaced because there is no pending_create_peers records. Instead it will be marked as a new peer, and then be marked as `already_exist_regions` in the following lines. If destroy happen after being marked as replaced, then it must have been marked as replaced. So destroy will [skip](https://github.com/tikv/tikv/pull/12386/files#diff-78aa13bfc4a34ffc7f594cd5e4b11ed46e2267b3fc650d43bb7371cbb1d76ee1R1025-R1027) deleting or updating any states, this panic is unreachable.

Reply:
Got it.